--- jam.h.orig	2021-04-25 12:09:14.000000000 +0200
+++ jam.h	2021-04-25 12:08:48.000000000 +0200
@@ -438,10 +438,6 @@
 # include <sys/types.h>
 # include <sys/stat.h>
 
-# ifdef OS_HPUX
-# include <unistd.h>
-# endif
-
 # ifndef OS_MPEIX
 # include <sys/file.h>
 # endif
@@ -452,6 +448,7 @@
 # include <signal.h>
 # include <string.h>
 # include <time.h>
+# include <unistd.h>
 
 # ifndef OS_QNX
 # include <memory.h>
