--- scan.h.orig	2021-04-25 12:02:53.000000000 +0200
+++ scan.h	2021-04-25 12:03:04.000000000 +0200
@@ -18,8 +18,8 @@
  *
  * The yymode() function is for the parser to adjust the lexicon of the
  * scanner.  Aside from normal keyword scanning, there is a mode to
- * handle action strings (look only for the closing }) and a mode to 
- * ignore most keywords when looking for a punctuation keyword.  This 
+ * handle action strings (look only for the closing }) and a mode to
+ * ignore most keywords when looking for a punctuation keyword.  This
  * allows non-punctuation keywords to be used in lists without quoting.
  *
  * 11/04/02 (seiwald) - const-ing for string literals
@@ -50,7 +50,7 @@
 int yylex();
 int yyparse();
 const char *yyfname();
-int yylineo();
+int yylineno();
 
 # define SCAN_NORMAL	0	/* normal parsing */
 # define SCAN_STRING	1	/* look only for matching } */
